### ISSUE TITLE
UWP project generation fails with com.unity.burst package installed

### ISSUE THREAD

**Issue by djohnsomsft** [microsoft/MixedRealityToolkit-Unity#6098]
In a blank 3D project with MRTK enabled setup to build for HoloLens ARM/ARM64:

1. Import the Burst package from Package Manager (project will build).
2. Import the MRTK Foundation package.
3. Attempt to generate the Visual Studio project.

Result:

![image](https://user-images.githubusercontent.com/27969160/65714963-05746e00-e051-11e9-83ee-627f1570ab6b.png)

Unity version: 2019.1.14f1
Windows SDK: 18362 (verified the referenced winmd is present)

This blocks usage of Unity's new Jobs and Entity-Component-System perf optimizations.

**Comment by djohnsomsft** [comment]
It appears this happens upon adding any reference to WinRT in C# code, even without MRTK itself. This may be a more general Unity issue.

**Comment by TautvydasZilys** [comment]
This is a Unity issue and we fixed it a while back. Can you update to the newest burst package version?

**Comment by djohnsomsft** [comment]
@TautvydasZilys I have updated to the latest preview I see in Package Manager at 1.1.3-preview.3 and the issue still reproduces. Is there a newer version I can access to try?

**Comment by djohnsomsft** [comment]
So after updating to 1.2.0-preview.5, while I am able to get a random WinRT API to compile, MRTK itself does not compile. It shows a different dependency not being included (which I verified is present in my SDK) - same repro steps just with the preview package:

![image](https://user-images.githubusercontent.com/27969160/65728742-b8a08f80-e070-11e9-8dd9-5c13c13e2c83.png)

**Comment by djohnsomsft** [comment]
Another data point: This reproduces in 2019.1.14f1 and 2019.2.5f1, but the build succeeds in 2019.3.0b4 with the Foundation package.

**Comment by TautvydasZilys** [comment]
Can you report this bug via a Unity bug reporter?

**Comment by djohnsomsft** [comment]
Done last night, though I didn't get an e-mail with the number (maybe I typo'd my e-mail or something?).

**Comment by TautvydasZilys** [comment]
Thanks, we will take a look.

**Comment by Deukhoofd** [comment]
The bug on the Unity issue tracker is marked as Wont Fix, and I'm still seeing this issue in Unity 2018.4.11f1, with Burst 1.2.0 preview 9. Does this mean that Unity is not planning to support Burst for MRTK?

**Comment by SheepJeffery** [comment]
same problem    UNITY 2021.3.0    Burst1.6

**Comment by keveleigh** [comment]
Although this is external, I'm reopening for tracking purposes as I believe I was seeing this when trying out GLTFast. Will continue investigating

**Comment by dudziakl** [comment]
@keveleigh  I can confirm that the issue is fixed finally with the Burst v1.7.2 package🎉
I have created a new project with Unity 2021.3.4f and imported UPR package that requires the Burst package. By default, it was 1.6.5 and when I tried to build UWP project I had an error:
> Mono.Cecil.AssemblyResolutionException: Failed to resolve assembly: 'Windows.Foundation, Version=255.255.255.255, Culture=neutral, PublicKeyToken=null'

I was lucky to find a great [post ](https://forum.unity.com/threads/unable-to-build-for-uwp.1264418/#post-8060261) about the manual update of the Burst package. Version 1.7.3 was not visible to me in Unity Package Manager.

I can confirm that only the Burst update was enough to be able to build the project and have the app working with HoloLens 2. 

![image](https://user-images.githubusercontent.com/12583808/181524010-499e6dd3-366b-4e2f-a8f8-d61d37cb4123.png)

Here is the note in the change log of Unity's Burst update https://docs.unity3d.com/Packages/com.unity.burst@1.7/changelog/CHANGELOG.html#172---2022-05-20
